rfcs/react-components/convergence/slot-render-as.md

```diff
@@ -0,0 +1,155 @@
+# RFC: Use `as` instead of `children` for slot render function
+
+Contributors: behowell
+
+## Summary
+
+This proposes modifying the slot render function API to pass a render function via the slot's `as` prop instead of the `children` prop. This has several benefits, the primary of which is that it allows the user to write a render function for a slot without overriding the default children of a slot.
```

Comment:
We will not have conflicts in `as` prop because we will define defaults under `state.components`, correct?

Reply:
Correct, yes.

Reply:
It would only be a problem if we wanted to use `slot.as` to store the default component for a slot, But the `as` prop can't be set to a React component, so it wouldn't be suitable for that.

The method of keeping track of the default component for a slot is out of scope for this RFC:
* As of today, it's solved via `state.components.SLOT`.
* In #27164, one of the proposals is to save it as `state.SLOT.componentType`.

Reply:
This wouldn't be a problem @layershifter, in any case the default component type will be available as an alternative property somehow.